scripts/check-agent-version-update.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+if [[ "${BRANCH_NAME}" == "" ]]; then
+    BRANCH_NAME=master;
+fi
+
+# Find the common ancestor between the current set of changes and the upstream branch, then see if any source code files
+# have changed in the agent or command packages.
+common_ancestor=$(git merge-base ${BRANCH_NAME}@{upstream} HEAD);
+if [[ "$(git diff --name-only ${common_ancestor} -- agent/**/*.go ':!agent/**/*_test.go' command/**/*.go ':!command/**/*_test.go')" == "" ]]; then
+    exit 0;
+fi
+
+# Get the list of commit revisions.
+committed_changes=$(git log --format=%H ${common_ancestor}..);
```

Comment:
I tested it with patches ([success](https://spruce.mongodb.com/version/5f89e0073e8e86687ab14e43/tasks), [failure](https://spruce.mongodb.com/version/5f89deca32f4176ae8b12071/tasks)) and [a dummy PR](https://github.com/evergreen-ci/evergreen/pull/4083) by changing an existing agent file and making a new directory (to test recursion) - both seem to correctly identify if the agent version has been updated.

I had to make a minor fix to the recursive pattern match since StackOverflow gave me an answer that seems incorrect (`agent/**/*.go` fails to match go files within agent).

Reply:
Maybe it's using [globstar](https://unix.stackexchange.com/a/49917)?

Reply:
I thought of that, but my local shell has globstar enabled and also behaves like this. The case in which `agent/**.go` did not work was when I made a subdirectory and put a new go source file in it. The case in which `agent/**/*.go` did not work was when I made a change to `agent/agent.go`.

Reply:
🤷‍♂️